rust_src/src/editfns.rs

```diff
@@ -317,6 +317,38 @@ pub fn preceding_char() -> EmacsInt {
     EmacsInt::from(buffer_ref.fetch_char(pos))
 }
 
+/// Return character in current buffer preceding position POS.
+/// POS is an integer or a marker and defaults to point.
+/// If POS is out of range, the value is nil.
+#[lisp_fn(min = "0")]
+pub fn char_before(pos: LispObject) -> Option<EmacsInt> {
+    let mut buffer_ref = ThreadState::current_buffer();
+    let pos_byte: isize;
+
+    if pos.is_nil() {
+        pos_byte = buffer_ref.pt_byte;
+    } else if let Some(m) = pos.as_marker() {
+        pos_byte = m.bytepos_or_error();
+        if pos_byte <= buffer_ref.begv_byte || pos_byte > buffer_ref.zv_byte {
+            return None;
+        }
+    } else {
```

Comment:
Why use coerce marker here? You just proved it is not a marker in the previous check? Either add a comment or just use `as_fixnum`.

Reply:
I just realized, that I used coerce for the error message.